packages/esm-form-entry-app/src/app/form-schema/form-schema.service.ts

```diff
@@ -80,6 +133,72 @@ export class FormSchemaService {
     return formSchema;
   }
 
+  private getFormTranslationsByUuidFromServer(formUuid: string) {
+    const formTranslations: ReplaySubject<any> = new ReplaySubject(1);
+    this.fetchFormTranslationUsingFormMetadata(formUuid).subscribe(
+      (translations: any) => {
+        // check whether formSchema has references b4 hitting getFormSchemaWithReferences
+        if (translations.referencedForms && translations.referencedForms.length > 0) {
+          this.getFormSchemaWithReferences(translations).subscribe(
+            (form: object) => {
+              formTranslations.next(form);
+            },
+            (err) => {
+              console.error(err);
+              formTranslations.error(err);
+            },
+          );
+        } else {
+          formTranslations.next({
+            form: translations,
+            referencedComponents: [],
+          });
+        }
+      },
+      (err) => {
+        console.error(err);
+        formTranslations.error(err);
+      },
+    );
+    return formTranslations;
+  }
+
+  private getFormTranslationResourceByUuidFromServer(translationResourceUuid: string) {
+    return this.formsResourceService.getFormClobDataByUuid(translationResourceUuid).toPromise();
+  }
```

Comment:
Does this function need to be here? It looks like it exists to turn an RXJS object into a Promise, but it's not clear what the advantage of doing that is...